test/FormFileSpec.js

```diff
@@ -0,0 +1,80 @@
+import { mount } from 'enzyme';
+import React from 'react';
+import FormFile from '../src/FormFile';
+
+describe('<FormFile>', () => {
+  it('should render correctly', () => {
+    let wrapper = mount(
+      <FormFile id="foo" name="foo" label="My label" className="my-file" />,
+    );
+
+    wrapper
+      .assertSingle('div.form-group.my-file')
+      .assertSingle('input[type="file"][name="foo"]');
+
+    wrapper
+      .assertSingle('label.form-file-label[htmlFor="foo"]')
+      .text()
+      .should.equal('My label');
+  });
+
+  it('should support isValid', () => {
+    mount(<FormFile isValid />).assertSingle('.is-valid');
+  });
+
+  it('should support isInvalid', () => {
+    mount(<FormFile isInvalid />).assertSingle('.is-invalid');
+  });
+
+  it('should support ref forwarding', () => {
+    class Container extends React.Component {
+      render() {
+        return (
+          <FormFile
+            ref={ref => {
+              this.input = ref;
+            }}
+          />
+        );
+      }
+    }
+
+    const instance = mount(<Container />).instance();
+
+    expect(instance.input.tagName).to.equal('INPUT');
+  });
+
+  it('should supports custom', () => {
+    const wrapper = mount(<FormFile custom label="My label" />);
+
+    wrapper
+      .assertSingle('div.custom-file')
+      .assertSingle('input.custom-file-input');
+
+    wrapper.assertSingle('label.custom-file-label');
+  });
+
+  it('should supports lang when custom', () => {
+    const wrapper = mount(<FormFile custom lang="en" label="My label" />);
+
+    expect(wrapper.prop('lang')).to.equal('en');
+  });
+
+  it('should supports buttonText when custom', () => {
```

Comment:
Is there any way to test with enzyme that the CSS content attribute that is used appears correctly?

In this case setting the buttonText
```
  <Form.File
    id="custom-file-translate"
    label="Custom file input"
    buttonText="Button text"
    custom
  />
```

Will  match this CSS
```
.custom-file-input~.custom-file-label[data-browse]::after {
    content: attr(data-browse);
}
```

And will thus output like this:
<img width="659" alt="Screenshot 2020-03-11 at 02 11 30" src="https://user-images.githubusercontent.com/322759/76372433-a758f000-633d-11ea-9410-1a2421e137e5.png">

We would probably like to test that "Button Text" in this case actually appears?

Reply:
I can answer this myself. EnzymeJS does not currently support testing ::after pseudo-elements. There is an open issue over at their GitHub: https://github.com/enzymejs/enzyme/issues/2102

Reply:
I can answer this myself. EnzymeJS does not currently support testing ::after pseudo-elements. There is an open issue over at their GitHub: enzymejs/enzyme#2102